cli/tools/test/reporters/pretty.rs

```diff
@@ -133,7 +133,7 @@ impl TestReporter for PrettyTestReporter {
   fn report_plan(&mut self, plan: &TestPlan) {
     self.summary.total += plan.total;
     self.summary.filtered_out += plan.filtered_out;
-    if self.parallel {
+    if self.parallel || plan.total == 0 {
```

Comment:
Maybe we should return here only total is 0 and filtered is non-0?

Reply:
I'm not sure what we'd gain from that. To me showing `running 0 tests` is always undesirable, but this might be personal preference.

Reply:
You had to change a bunch of tests assertions - they were printing that we're running 0 tests from certain files - I think this is desirable if you don't have a filter as it becomes obvious that you have a test file that has no test cases (plus these files were testing auto-discovery based on various config files). Otherwise it's easy for the file to rot. I think it does make sense no to print file names when you do apply a filter.

Reply:
Right, that makes sense 👍